code/_core/obj/structure/interactive/mountable/browning.dm

```diff
@@ -0,0 +1,297 @@
+	plane = PLANE_MOB
+	layer = LAYER_MOB_ABOVE
+
+	density = TRUE
+
+	bullet_block_chance = 15	//it's pretty big and chairs literally have 25
+
+	var/obj/item/weapon/ranged/bullet/magazine/misc/browning/stored_weapon
+	var/obj/item/magazine/stored_magazine
+	var/obj/item/browning_handle/stored_handle
+
+	stored_weapon = /obj/item/weapon/ranged/bullet/magazine/misc/browning
+
+	var/obj/structure/interactive/mountable/browning/direction
+
+	var/is_container = TRUE
+
+//imagine actually knowing how to code in byond
+
+/obj/item/browning_handle
+	name = "browning handles"
+	desc = "Keep 'em steady."
+	desc_extended = "You'll need to wield them with both hands to fire the gun properly."
+	icon = 'icons/obj/item/browning_handle.dmi'
+
+	var/obj/structure/interactive/mountable/browning/linked_gun
+
+	should_save = FALSE
+	throwable = FALSE
+	can_wield = TRUE
+
+	on_drop(var/obj/hud/inventory/old_inventory,var/atom/new_loc,var/silent=TRUE)
+		if(!istype(new_loc, /obj/hud/inventory/organs/)) //switching hands drops and fucks it
+			wielded = FALSE
+			if(old_inventory.child_inventory) //from weapons, deals with dropping the wield
+				old_inventory.child_inventory.parent_inventory = null
+				old_inventory.child_inventory.update_sprite()
+				old_inventory.child_inventory = null
+			if(linked_gun == null)
+				qdel(src)
+			else
+				src.force_move(linked_gun)
+			return TRUE
+		return ..()
+
+	size = 99
+
+	weight = 2
+
+//taken from chairs, sits down if you move onto it, but only once
+/obj/structure/interactive/mountable/browning/Crossed(atom/movable/O)
+
+	. = ..()
+
+	if(is_living(O))
+		CALLBACK("sit_down_\ref[src]",SECONDS_TO_DECISECONDS(1),src,.proc/mount_your_ass,O)
+
+
+/obj/structure/interactive/mountable/browning/proc/mount_your_ass(var/mob/living/L)
+
+	if(L.loc != src.loc)
+		return FALSE
+
+	L.set_dir(src.dir)
+
+	return TRUE
+
+/obj/structure/interactive/mountable/browning/proc/mag_check(var/structure/interactive/mountable/browning/M)
+	if(stored_magazine)
+		if(length(stored_magazine.stored_bullets) < 1)
+			icon_state = "empty"
+		else
+			icon_state = "loaded"
+		return TRUE
+	else
+		icon_state = "unloaded"
+
+	return TRUE
+
+//taken from barricades, gun sprite shifted to position around player sprites
+/obj/structure/interactive/mountable/browning/PostInitialize()
+
+	update_sprite()
+
+	. = ..()
+
+/obj/structure/interactive/mountable/browning/update_sprite()
+
+	mag_check(src)
+
+	if(dir == NORTH)
+		pixel_x = 0
+		pixel_y = 16
+		layer = LAYER_MOB_BELOW
+	else if(dir == EAST)
+		pixel_x = 16
+		pixel_y = -3
+	else if(dir == SOUTH)
+		pixel_x = 0
+		pixel_y = -17
+	else if(dir == WEST)
+		pixel_x = -16
+		pixel_y = -3
+
+	. = ..()
+
+//make 'em handles
+//from turret
+/obj/structure/interactive/mountable/browning/Finalize()
+
+	stored_weapon = new stored_weapon(src)
+	INITIALIZE(stored_weapon)
+	GENERATE(stored_weapon)
+	FINALIZE(stored_weapon)
+
+	stored_weapon.linked_gun = src
+
+	stored_handle = new /obj/item/browning_handle(src)
```

Comment:
CREATE()'s defined as this:
#define CREATE(I,desired_loc) var/datum/D = new I(desired_loc);INITIALIZE(D);GENERATE(D);FINALIZE(D)
Are you sure it can make /obj/ects? I'm getting tons of undefined vars.